addons/a11y/src/a11yHighlight.ts

```diff
@@ -0,0 +1,44 @@
+import { document } from 'global';
+import addons from '@storybook/addons';
+import { EVENTS, HIGHLIGHT_STYLE_ID } from './constants';
+
+if (module && module.hot && module.hot.decline) {
+  module.hot.decline();
+}
+
+interface HighlightInfo {
+  /** html selector of the element */
+  elements: string[];
+  color: string;
+}
+
+const channel = addons.getChannel();
+
+const higlighted = (color: string) => `
+  outline: 2px dashed ${color};
+  outline-offset: 2px;
+  box-shadow: 0 0 0 6px rgba(255,255,255,0.6);
+}
+`;
+
+const highlight = (infos: HighlightInfo) => {
```

Comment:
Let's extract into a different module and re-sue it in a story so we can test this visually.